File: src/Javascript/greedy/55.跳跃游戏.js
/*
 * @lc app=leetcode.cn id=55 lang=javascript
 *
 * [55] 跳跃游戏
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let maxPosition = 0;
    for (let i = 0; i < nums.length; i++) {
        if (i > maxPosition) return false;
        maxPosition = Math.max(maxPosition, nums[i] + i);
        
    }
    return true;
};
// @lc code=end

File: src/Java/Mirosoft/55.跳跃游戏.java
/*
 * @lc app=leetcode.cn id=55 lang=java
 *
 * [55] 跳跃游戏
 */

// @lc code=start
class Solution {
    public boolean canJump(int[] nums) {
        int maxJump = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (i > maxJump) return false;
            maxJump = Math.max(nums[i] + i, maxJump);
        }
        return true;
    }
}
// @lc code=end

